polyglot/piranha/src/cleanup_rules/go/rules.toml

```diff
@@ -0,0 +1,547 @@
+#  if (true) { doSomething(); }
+# After :
+#  { doSomething(); }
+#
+# Before :
+#  if (true) { doSomething(); } else { doSomethingElse();}
+# After :
+#  { doSomething(); }
+#
+[[rules]]
+groups = ["if_cleanup"]
+name = "simplify_if_statement_true"
+query = """
+(
+    (if_statement
+        condition : (
+            [
+                (true)
+                (parenthesized_expression (true))
+            ]
+        )
+        consequence : ((block) @consequence)
+    ) @if_statement
+)
+"""
+replace = "@consequence"
+replace_node = "if_statement"
+
+# Before :
+#  if (true) { doSomething(); } else { doSomethingElse();}
+# After :
+#  { doSomethingElse(); }
+#
+# Before :
+#  if (true) { doSomething(); }
+# After :
+#
+[[rules]]
+groups = ["if_cleanup"]
+name = "simplify_if_statement_false"
+query = """
+(
+    (if_statement
+        condition : (
+            [
+                (false)
+                (parenthesized_expression (false))
+            ]
+        )
+        consequence : (_)
+        alternative: ((_) @alternative) ?
+    ) @if_statement
+)
+"""
+replace = "@alternative"
+replace_node = "if_statement"
+
+# Before :
+#  {
+#     someStepsBefore();
+#     {
+#        someSteps();
+#     }
+#     someStepsAfter();
+#  }
+# After :
+#  {
+#     someStepsBefore();
+#        someSteps();
+#     someStepsAfter();
+#  }
+#
+# Note that we need to tag basically all nodes here.
+# Including not so obvious ones: @outer.stmt_list and @outer.block
+[[rules]]
+name = "remove_unnecessary_nested_block"
+query = """
+(
+    (block
+        (statement_list
+            (_)* @pre
+            (block
+                (statement_list) @nested.statements
+            ) @nested.block
+            (_)* @post
+        ) @outer.stmt_list
+    ) @outer.block
+)
+"""
+replace = "@nested.statements"
+replace_node = "nested.block"
+
+# This rule is not deleting multiple statements after return.
+# `(statement)+ @post` is only matching the first statement
+# Thus, we are using a self-edge to keep removing statements.
+#
+[[rules]]
+name = "delete_statement_after_return"
+query = """
+(
+    (block
+        (statement_list
+            ((statement)* @pre)
+            ((return_statement) @r)
+            ((statement)+ @post)
+        ) @stmt_list
+    ) @b
+)
+"""
+replace = ""
+replace_node = "post"
+
+# TODO: rules and edges for "if with short statement"
+# collect examples and write tests for it
+# https://go.dev/tour/flowcontrol/6
+# https://go.dev/doc/effective_go#if
+
+# TODO: we need a different rule for `nil != err`
+# left: [
+#     (identifier) @id
+#     (nil)
+# ]
+# making `left` and `right` options could lead to `(identifier) != (identifier)`
+
+# Parent matches the current node and up to FOUR nodes above.
+# This rule is a best attempt to remove an assignment that is not reused later.
+[[rules]]
+name = "delete_parent_assignment"
+query = """
+(
+    [
+        (short_var_declaration
+            left: (expression_list
+                .
```

Comment:
why this `.` ? Can we do without them ?

Reply:
The `.` is to make sure that we have only one identifier on the LHS.
If we don't use `.`, the query will match both `b` and `err` in the following:

```go
b, err := true, nil
```

Tests still pass if you prefer to remove the `.`